webdriver/tests/bidi/browsing_context/__init__.py

```diff
@@ -34,15 +41,22 @@ def assert_browsing_context(
     assert info["url"] == url
 
 
-def assert_navigation_info(event, context, url):
-    assert "context" in event
-    assert isinstance(event["context"], str)
-    assert event["context"] == context
+def assert_navigation_info(event, expected_navigation_info):
+    recursive_compare(
+        {
+            "context": any_string,
+            "navigation": any_string_or_null,
+            "timestamp": any_int,
+            "url": any_string,
+        },
+        event,
+    )
+
+    if "context" in expected_navigation_info:
```

Comment:
I'd just call here `recursive_compare` again:
```
recursive_compare(
        expected_navigation_info,
        event
    )
```

Reply:
We don't want to check every property for every test, eg. setting up the timestamp is intensive and as such it's enough to have a single test feeding in the timespan to assert. If we find a way to make it easier in the future we can change it for sure.